source/gloperate/source/loaders/GlrawTextureLoader.cpp

```diff
@@ -0,0 +1,165 @@
+
+#include <gloperate/loaders/GlrawTextureLoader.h>
+
+#include <algorithm>
+
+#include <cppassist/fs/FilePath.h>
+#include <cppassist/fs/readfile.h>
+#include <cppassist/fs/RawFile.h>
+#include <cppassist/fs/DescriptiveRawFile.h>
+
+#include <cppexpose/variant/Variant.h>
+
+#include <glbinding/gl/enum.h>
+
+#include <globjects/Texture.h>
+
+#include <gloperate/base/RawFileNameSuffix.h>
+
+
+CPPEXPOSE_COMPONENT(GlrawTextureLoader, gloperate::AbstractLoader)
+
+
+GlrawTextureLoader::GlrawTextureLoader(gloperate::Environment * environment)
+    : gloperate::Loader<globjects::Texture>(environment)
+{
+    m_extensions.push_back(".raw");
+    m_extensions.push_back(".glraw");
+
+    m_types.push_back("raw images (*.raw)");
+    m_types.push_back("glraw images (*.glraw)");
+}
+
+GlrawTextureLoader::~GlrawTextureLoader()
+{
+}
+
+bool GlrawTextureLoader::canLoad(const std::string & ext) const
+{
+    // Check if file type is supported
+    return std::find(m_extensions.begin(), m_extensions.end(), "." + ext) != m_extensions.end();
+}
+
+std::vector<std::string> GlrawTextureLoader::loadingTypes() const
+{
+    // Return list of supported file types
+    return m_types;
+}
+
+std::string GlrawTextureLoader::allLoadingTypes() const
+{
+    // Compose list of all supported file extensions
+    std::string allTypes;
+    for (unsigned int i = 0; i < m_extensions.size(); ++i) {
+        if (i > 0) allTypes += " ";
+        allTypes += "*." + m_extensions[i].substr(1);
+    }
+
+    // Return supported types
+    return allTypes;
+}
+
+globjects::Texture * GlrawTextureLoader::load(const std::string & filename, const cppexpose::Variant & /*options*/, std::function<void(int, int)> /*progress*/) const
+{
+    globjects::Texture * texture = nullptr;
+
+    cppassist::FilePath filePath(filename);
+    if (filePath.extension() == "glraw")
+        texture = loadGLRawImage(filename);
+    else if (filePath.extension() == "raw")
+        texture = loadRawImage(filename);
+
+    return texture;
+}
+
+globjects::Texture * GlrawTextureLoader::loadGLRawImage(const std::string & filename) const
+{
+    cppassist::DescriptiveRawFile rawFile;
+    if (!rawFile.load(filename))
+        return nullptr;
+
+    const int w = rawFile.intProperty("width");
+    const int h = rawFile.intProperty("height");
+
+    //TODO this should be a unique ptr like this:
+    //auto texture = globjects::Texture::createDefault(gl::GL_TEXTURE_2D);
+    globjects::Texture * texture = new globjects::Texture(gl::GL_TEXTURE_2D);

```

Comment:
Use ```globjects::Texture::createDefault(gl::GL_TEXTURE_2D);``` instead as this call initializes with default sampling parameters.
Using the current interface, a texture is not ready to use in shaders as sampler information are missing.